strapi/strapi
# Fix: typescript build cleaning and type generation

### What does it do?

- cleans the dist directory before each build
- `develop` command builds ignoring errors to allow types to generate
- adds Math.ceil() on all displayed timings instead of showing float decimals

Note:  This PR does not change types generation on startup, meaning that manual schema changes still require a `yarn strapi ts:generate-types` before starting the server

### Why is it needed?

Currently using the CTB with a typescript project, creating a new content type causes the server to become unable to start up again without manually running ts:generate-types

Additionally, the dist dir was not being cleaned, meaning that old content types would appear in the CTB until the dist folder was cleaned manually.

### How to test it?

Create and delete new content types with the CTB. Now it should work and the server should restart, new content types should appear, and old deleted content types should no longer appear.

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/18738
